operations/document-compression-updater/updater.py

```diff
@@ -0,0 +1,271 @@
+            time.sleep(appConfig['waitPeriod'])
+        else :
+            print("No updates in batch")
+            allDone = True
+            break
+        
+
+                
+    client.close()
+    
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Data Generator')
+
+    parser.add_argument('--uri',required=True,type=str,help='URI (connection string)')
+    parser.add_argument('--database',required=True,type=str,help='Database')
+    parser.add_argument('--collection',required=True,type=str,help='Collection')
+    parser.add_argument('--file-name',required=False,type=str,default='compressor',help='Starting name of the created log files')
+    parser.add_argument('--update-field',required=False,type=str,default='temp_field_for_compressor',help='Field used for updating an existing document. This should not conflict with any fieldname you are already using ')
+    parser.add_argument('--wait-period',required=False,type=int,default=60,help='Number of seconds to wait between each batch')
+    parser.add_argument('--batch-size',required=False,type=int,default=5000,help='Number of documents to update in a single batch')
+
+    
+
+    args = parser.parse_args()
+    
+    appConfig = {}
+    appConfig['uri'] = args.uri
+    appConfig['numInsertProcesses'] = 1 #int(args.processes)
+    appConfig['databaseName'] = args.database
+    appConfig['collectionName'] = args.collection
+    appConfig['updateField'] = args.update_field
+    appConfig['batchSize'] = int(args.batch_size)
+    appConfig['waitPeriod'] = int(args.wait_period)
+    appConfig['logFileName'] = "{}.log".format(args.file_name)
+
+    
+
+    setUpdata = setup(appConfig)
+    
+    appConfig['numExistingDocuments'] = setUpdata["numExistingDocuments"]  
+    appConfig['maxObjectIdToTouch'] = setUpdata["maxObjectIdToTouch"]  
+    appConfig['lastScannedObjectId'] = setUpdata["lastScannedObjectId"]     
+    appConfig['numDocumentsUpdated'] = setUpdata["numDocumentsUpdated"]   
+
+    
+    deleteLog(appConfig)
+    
+    printLog('---------------------------------------------------------------------------------------',appConfig)
+    for thisKey in sorted(appConfig):
+        if (thisKey == 'uri'):
+            thisUri = appConfig[thisKey]
+            thisParsedUri = pymongo.uri_parser.parse_uri(thisUri)
+            thisUsername = thisParsedUri['username']
+            thisPassword = thisParsedUri['password']
+            thisUri = thisUri.replace(thisUsername,'<USERNAME>')
+            thisUri = thisUri.replace(thisPassword,'<PASSWORD>')
+            printLog("  config | {} | {}".format(thisKey,thisUri),appConfig)
+        else:
+            printLog("  config | {} | {}".format(thisKey,appConfig[thisKey]),appConfig)
+    printLog('---------------------------------------------------------------------------------------',appConfig)
+    
+    mp.set_start_method('spawn')
+    q = mp.Manager().Queue()
```

Comment:
Why is this script including what appears to be multiprocessing capability yet it isn't multiprocessing capable? You are not going to be able to go fast without concurrency.

Reply:
Since this is updates we are keeping it single process to be on the safer side and not issue a lot of updates at a time. During the initial design discussion we agreed on keeping the code capable of multi processing in future without major overhaul, but default to one process in this release